wpimath/src/main/java/edu/wpi/first/math/geometry/Pose2d.java

```diff
@@ -238,6 +242,29 @@ public Twist2d log(Pose2d end) {
     return new Twist2d(translationPart.getX(), translationPart.getY(), dtheta);
   }
 
+  /**
+   * Returns the nearest Pose2d from a list of poses. If two or more poses in the list have the same
+   * distance from this pose, return the one with the closest rotation component.
+   *
+   * @param poses The list of poses to find the nearest.
+   * @return The nearest Pose2d from the list.
+   */
+  public Pose2d nearest(List<Pose2d> poses) {
+    return Collections.min(
+        poses,
+        Comparator.comparing(
+                (Pose2d other) -> {
+                  System.out.println(this.getTranslation().getDistance(other.getTranslation()));
+                  return this.getTranslation().getDistance(other.getTranslation());
+                })
+            .thenComparing(
+                (Pose2d other) -> {
+                  return Math.abs(
+                      MathUtil.angleModulus(
+                          this.getRotation().minus(other.getRotation()).getRadians()));
```

Comment:
Isn't the resulting Rotation2d from subtraction already in (-π,π]?

Reply:
Yeah, it uses the sin and cos components to do the operation. I'll remove the angleModulus.